### PR TITLE
client: temp files shouldn't be writable

### PR DESCRIPTION
**Pull request by mcpherrinm**
Don't want world-writable files.

### CODE REVIEW

File: client/sharkey_client.go [hunk 1]
```diff
@@ -155,12 +155,12 @@ func (c *context) enroll(hostKey string, signedCert string) {
...
 	}
 	defer os.Remove(tmp.Name())
-	err = os.Chmod(tmp.Name(), 0666)
+	err = os.Chmod(tmp.Name(), 0644)
 	if err != nil {
 		log.Printf("Error calling chmod on %s: %s\n", tmp.Name(), err)
 		return
 	}
-	err = ioutil.WriteFile(tmp.Name(), body, 0666)
+	err = ioutil.WriteFile(tmp.Name(), body, 0544)
```
Comment (tbosserman) [inline client/sharkey_client.go]: Don't you mean 0644 here? 0544 is executable but not writable!

Comment (mcpherrinm) [reply client/sharkey_client.go]: sigh, I do.

Comment (mcpherrinm) [reply client/sharkey_client.go]: I should add a test around this since I fat-fingered it once already.

Comment (mbyczkowski) [reply client/sharkey_client.go]: Yeah, a test would be great!

**Pull request merged**